std/getopt.d

```diff
@@ -1773,3 +1773,37 @@ void defaultGetoptFormatter(Output)(Output output, string text, Option[] opt)
     getopt(args, "f|flag", "Boolean", &flag);
     assert(flag);
 }
+
+@system unittest  // Delegates as callbacks
```

Comment:
I tried `@safe` initially, but it fails. Probably why most of the other unit tests are `@system`.  The error with `@safe`:
```
std/getopt.d(1799): Error: @safe function 'std.getopt.__unittestL1777_33' cannot call @system function 'std.getopt.getopt!(string, string, void delegate() pure nothrow @nogc @safe, string, string, void delegate(string option) pure nothrow @nogc @safe, string, string, void delegate(string option, string value), string, string, void delegate() pure nothrow @nogc @safe, string, string, void delegate(string option) pure nothrow @nogc @safe, string, string, void delegate(string option, string value)).getopt'
```

I should have mentioned in the initial description - One rationale for adding delegates to unit tests is that delegates and functions have separate tests in the getopt code. Doesn't show up in code coverage reports because the tests are on the same code line.